### ISSUE TITLE
Crash and assertion on the Mac in recent commit

### ISSUE THREAD

**Issue by eofster** [pjsip/pjproject#2453]
The commit c2d7c7616fc5240d7c6d56c454ba8f0acf9ac438 introduced crash on the Mac:
```
* thread #6, stop reason = EXC_BAD_ACCESS (code=1, address=0x8)
pj_list_find_node(list=0x00007fff90443088, node=0x0000000100358a10) at list_i.h:92:21
pjsip_endpt_register_module(endpt=0x00007fff90442f30, mod=0x0000000100358a10) at sip_endpoint.c:170:5
pjsip_inv_usage_init(endpt=0x00007fff90442f30, cb=0x000070000f5afb60) at sip_inv.c:824:14
pjsua_call_subsys_init(cfg=0x000070000f5b1270) at pjsua_call.c:222:14
pjsua_init(ua_cfg=0x000070000f5b1270, log_cfg=0x000070000f5b1240, media_cfg=0x000070000f5b0f50) at pjsua_core.c:1170:14
```
`p` is `NULL` here:
https://github.com/pjsip/pjproject/blob/55cd705114c493791b65cc5e3e5ca37d2d285261/pjlib/include/pj/list_i.h#L92

In the latest commit in master 55cd705114c493791b65cc5e3e5ca37d2d285261 it is, however, an assertion:
```
Assertion failed: (endpt), function pjmedia_codec_register_audio_codecs, file ../src/pjmedia-codec/audio_codecs.c, line 39.
* thread #5, stop reason = signal SIGABRT
pjmedia_codec_register_audio_codecs(endpt=0x0000000000000000, c=0x000070000b47cb38) at audio_codecs.c:39:5
pjsua_aud_subsys_init at pjsua_aud.c:300:14
pjsua_media_subsys_init(cfg=0x000070000b47df50) at pjsua_media.c:96:14
pjsua_init(ua_cfg=0x000070000b47e270, log_cfg=0x000070000b47e240, media_cfg=0x000070000b47df50) at pjsua_core.c:1194:14
```

**Comment by sauwming** [comment]
I tried it on a local Mac but couldn't reproduce the issue. The test build on CI machine also seems fine (https://github.com/pjsip/pjproject/runs/737365484?check_suite_focus=true). For the later issue there's a PR to fix a bug in #2454, but your stack trace seems to be unrelated to the issue. In fact, both stack traces don't seem related to the issue since it happens so early in pjsua_init().

So my suggestion is to try a clean build, and if the issue persists, please let us know your configure parameters as well as config_site.h content, if not using the default.

**Comment by eofster** [comment]
I’m sorry for the false alarm, cleaning resolves the problem. Thanks for testing it.
